pkg/noderesourcetopology/cache/cache.go

```diff
@@ -28,7 +28,9 @@ type Interface interface {
 	// Over-reserved resources are the resources consumed by pods scheduled to that node after the last update
 	// of NRT pertaining to the same node, pessimistically overallocated on ALL the NUMA zones of the node.
 	// The pod argument is used only for logging purposes.
-	GetCachedNRTCopy(nodeName string, pod *corev1.Pod) *topologyv1alpha1.NodeResourceTopology
+	// Returns a boolean to signal the caller if the NRT data is clean. If false, then the node is possibly
+	// tainted - so it should be ignored or handled differently by the caller.
```

Comment:
After reading through the logic, I realized `tainted` here doesn't mean the k8s `Taint`... so maybe reword it a bit to avoid confusion, or add a comment saying it's not the k8s Taint.

Reply:
good point, will rename. Also good point about multiple profiles: in my testing (and in the usecases we are covering) we will have a secondary scheduler with a single profile, this is why I missed this before.
A way to fix is to turn the environment variable value into a comma separate list of profile names. This is pretty ugly and duplicates information, but AFAICT should work.

Reply:
Addressed the multiple profile comments. I'm not too happy about the UX but this is the best I can think of atm. I'd like to keep iterating here in the next revision cycles.